v2/managed-io-to-managed-io/README.md

```diff
@@ -0,0 +1,231 @@
+
+| Parameter | Description |
+|-----------|-------------|
+| sourceConnectorType | The type of Managed I/O connector to use as source. Supported values: ICEBERG, ICEBERG_CDC, KAFKA, BIGQUERY. |
+| sourceConfig | JSON configuration for the source Managed I/O connector. The configuration format depends on the connector type. |
+| sinkConnectorType | The type of Managed I/O connector to use as sink. Supported values: ICEBERG, KAFKA, BIGQUERY. Note: ICEBERG_CDC is only available for reading. |
+| sinkConfig | JSON configuration for the sink Managed I/O connector. The configuration format depends on the connector type. |
+| streaming | (Optional) Whether to run the pipeline in streaming mode. This is only supported for certain connector combinations. Default is false (batch mode). |
+
+## Supported Managed I/O Connectors
+
+### Source Connectors
+
+#### ICEBERG
+Reads from Apache Iceberg tables.
+
+**Example Configuration:**
+```json
+{
+  "table": "my_catalog.my_database.my_table",
+  "catalog_name": "my_catalog",
+  "catalog_properties": {
+    "warehouse": "gs://my-bucket/warehouse",
+    "catalog-impl": "org.apache.iceberg.gcp.biglake.BigLakeCatalog"
+  }
+}
+```
+
+#### ICEBERG_CDC
+Reads change data capture (CDC) streams from Apache Iceberg tables.
+
+**Example Configuration:**
+```json
+{
+  "table": "my_catalog.my_database.my_table",
+  "catalog_name": "my_catalog",
+  "streaming": true,
+  "starting_strategy": "latest"
+}
+```
+
+#### KAFKA
+Reads from Apache Kafka topics.
+
+**Example Configuration:**
+```json
+{
+  "bootstrap_servers": "localhost:9092",
+  "topic": "input-topic",
+  "format": "JSON"
+}
+```
+
+#### BIGQUERY
+Reads from Google BigQuery tables.
+
+**Example Configuration:**
+```json
+{
+  "table": "project:dataset.table",
+  "query": "SELECT * FROM `project.dataset.table` WHERE date >= '2024-01-01'"
+}
+```
+
+### Sink Connectors
+
+#### ICEBERG
+Writes to Apache Iceberg tables.
+
+**Example Configuration:**
+```json
+{
+  "table": "my_catalog.my_database.output_table",
+  "catalog_name": "my_catalog",
+  "catalog_properties": {
+    "warehouse": "gs://my-bucket/warehouse",
+    "catalog-impl": "org.apache.iceberg.gcp.biglake.BigLakeCatalog"
+  }
+}
+```
+
+#### KAFKA
+Writes to Apache Kafka topics.
+
+**Example Configuration:**
+```json
+{
+  "bootstrap_servers": "localhost:9092",
+  "topic": "output-topic",
+  "format": "JSON"
+}
+```
+
+#### BIGQUERY
+Writes to Google BigQuery tables.
+
+**Example Configuration:**
+```json
+{
+  "table": "project:dataset.output_table"
+}
+```
+
+## Usage Examples
+
+### Example 1: Kafka to BigQuery
+
+```bash
+gcloud dataflow flex-template run "managed-io-kafka-to-bigquery" \
+    --template-file-gcs-location="gs://dataflow-templates/latest/flex/Managed_IO_to_Managed_IO" \
+    --region="us-central1" \
+    --parameters="sourceConnectorType=KAFKA" \
+    --parameters="sourceConfig={\"bootstrap_servers\": \"localhost:9092\", \"topic\": \"input-topic\", \"format\": \"JSON\"}" \
```

Comment:
Hand crafting JSON is painful. Can we add a util or refer to a util to generate this ?

Reply:
I have a list of small features we can do later: 1) support the json gs url 2) support UDF 3) DLQ. Let us keep this initial PR simple.

Reply:
and 4) more tests with different IOs.

Reply:
https://github.com/GoogleCloudPlatform/DataflowTemplates/issues/2590 tracks this.